lib/src/participant/remote.dart

```diff
@@ -167,15 +166,13 @@ class RemoteParticipant extends Participant<RemoteTrackPublication> {
       }
     }
 
-    // notify listeners when it's not a new participant
-    if (hadInfo) {
-      for (final pub in newPubs) {
-        final event = TrackPublishedEvent(
-          participant: this,
-          publication: pub,
-        );
-        [events, room.events].emit(event);
-      }
+    // always emit events for new publications, Room will not forward them unless it's ready
+    for (final pub in newPubs) {
+      final event = TrackPublishedEvent(
+        participant: this,
+        publication: pub,
+      );
+      [events, room.events].emit(event);
```

Comment:
I don't believe we want to do this here. TrackPublished events should not be emitted before the Room is connected.

When a room is already connected
* fire `TrackPublished`

When a room is not yet connected
* do *not* fire `TrackPublished`

Reply:
If the room here is not connected, do we need to cache the TrackPublished event until room is reconncted and then emit it?

Reply:
I don't think we need to cache it

Reply:
the event is only fired for tracks that are published after the participant has joined

Reply:
so, here we need to confirm that the `subscriber.pc` is connected, then the event can be emitted, is this correct?

Reply:
not exactly.. to be clear, when I say "connected", it's referring to Room.state == connected. This is the contract we have with developers

Reply:
ok, I'm trying to figure out the timing,  there are two `_updateConnectionState(ConnectionState.connected);` in core/engine.dart that let me know the room is connected. 
so, no more questions